guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt

```diff
@@ -0,0 +1,18 @@
+package com.spectralogic.kotlinutils.core
+/*
+ Composible curry function for reducing the arity of a given function
+ */
+
+fun <A,B> ((A) -> B).curry(a: A): B = this.invoke(a)
```

Comment:
Include some Javadoc for each of these curry functions to better explain what the different versions do.